### PR TITLE
[Resource] Upgraded ResourceDeleteSubscriber

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

Now it'll only handle admin's delete action of sylius resources exclusively.
No more random catching of failed updates, or any other crap like that.

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php [hunk 1]
```diff
@@ -161,4 +172,40 @@ private function isHtmlRequest(Request $request)
...
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return bool
+     */
+    private function isMethodDelete(Request $request)
+    {
+        return 'DELETE' === $request->getMethod();
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php]: `Request:: METHOD_DELETE`

File: src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php [hunk 2]
```diff
@@ -161,4 +172,40 @@ private function isHtmlRequest(Request $request)
...
+     *
+     * @return bool
+     */
+    private function isAdminSection(array $syliusParameters)
+    {
+        return
+            array_key_exists('section', $syliusParameters) ?
+            'admin' === $syliusParameters['section'] :
+            false
+        ;
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php]:
```php
return array_key_exists('section', $syliusParameters) && 'admin' === $syliusParameters['section']
```

File: src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php [hunk 3]
```diff
@@ -161,4 +172,40 @@ private function isHtmlRequest(Request $request)
...
+    {
+        return 'DELETE' === $request->getMethod();
+    }
+
+    /**
+     * @param string $route
+     *
+     * @return bool
+     */
+    private function isSyliusResource($route)
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php]: isSylius**Route**

File: src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php [hunk 4]
```diff
@@ -161,4 +172,40 @@ private function isHtmlRequest(Request $request)
...
+    {
+        $routeArray = explode('_', $route);
+
+        return false !== $routeArray ? 'sylius' === $routeArray[0] : false;
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php]:
```php
return !empty($routeArray) && 'sylius' === $routeArray[0]
```

Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/EventListener/ResourceDeleteSubscriber.php]: `return 0 === strpos($route, 'sylius');`

**Comment by michalmarcinkowski** [comment]
Thanks Janek!

**Comment by pjedrzejewski** [comment]
This listener is referencing "admin" section and its purpose is to display nice flashes in admin section so hm... shouldn't it be in AdminBundle? ;)

**Pull request merged**